Change wording of relation queries

wielinde opened the pull request:
Multiple fixes and improvements for form configuration
- [x] Fix old dev and make old feature specs pass
- [x] Add styles for disabled tabs, which got lost when updating the styles for tabs
- [x] Change layout and wording when adding a relation filter to a query
- [x] Save form group names on form save even when the edit in place field still is open https://community.openproject.com/wp/32189

wielinde committed:
Fix 'ExpressionChangedAfterItHasBeenCheckedError' in form configuration

wielinde committed:
Fix: add styles for disabled tabs

wielinde committed:
WIP

openprojectci commented:
<!--
  0 Errors
  2 Warnings: <a href="https://github.com/op...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/64ce0071c0e876c5f7108df7c576f6627e67280f/frontend/src/app/components/wp-table/configuration-modal/wp-table-configuration-relation-selector.ts#L13">frontend/src/app/components/wp-table/configuration-modal/wp-table-configuration-relation-selector.ts#L13</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/opf/openproject/blob/64ce0071c0e876c5f7108df7c576f6627e67280f/frontend/src/app/modules/admin/types/type-form-configuration.component.ts#L32">frontend/src/app/modules/admin/types/type-form-configuration.component.ts#L32</a> - Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

oliverguenther committed:
Trigger form submission through mousedown

It's the only way we can capture Firefox broken blur/click handling in
this case

ulferts commented on frontend/src/app/components/wp-table/configuration-modal/wp-table-configuration-relation-selector.ts:
```diff
@@ -35,21 +35,22 @@ export class WpTableConfigurationRelationSelectorComponent implements OnInit  {
+    requires:      this.I18n.t('js.relation_labels.required'),
+    required:    this.I18n.t('js.relation_labels.requires'),
+    partof:        this.I18n.t('js.relation_labels.includes'),
+    includes:      this.I18n.t('js.relation_labels.partof')
```
This will invalidate all translations existing in languages translated on crowdin. I am by no means against changing the keys, but we should probably not be doing it this short before a release and if we do, we need to have the necessary translations added.

oliverguenther replied:
The `relation_labels.*` translations all exist for quite some time, only the special "attribute_groups" have been removed so I think this is safe with the exception of the one added line in https://github.com/opf/openproject/pull/8037/files#diff-92491e049486bcd60575ad84794538e1R840

ulferts replied:
You are right of course @oliverguenther. I mistakenly interpreted this to be the source (yml), so nevermind.

oliverguenther reviewed (approved):
The changes LGTM content wise and wording wise. The behavior is still the same so we won't break any exsting queries. 

Good to merge from my side. Let me know if this needs any additional work
